### ISSUE TITLE
"signed integer is greater than maximum" when creating cache

### ISSUE THREAD

**Issue by marcin-github** [elmystico/EnhanceIO#1]
I'm getting error reported here https://github.com/stec-inc/EnhanceIO/issues/77 . If you think that there is correct fix you can commit to your repo. What is the difference between your fork and  lanconnected/EnhanceIO? it looks that no new change appeared in your repository.

**Comment by elmystico** [comment]
Cześć Marcin
Po angielsku, gdyby miało się innym przydać:

I haven't integrated all changes yet into master branch. My current development is udev_systemd.
There you can find working (tested but messy) code.
Keep in mind that i've tested it in debian/systemd/initramfs-tools only. But should work on any systemd-based system with proper configuration- have necessary modules for drives and byte-compiled eio_cli in your initrd.
Test that in your environment and let me know if it works.

Soon I'll integrate code I'm working on now and push into master branch (two-three days more I guess).

regards

**Comment by marcin-github** [comment]
Cześć, nie zwróciłem uwagi na informacje na Twoim profilu:)

I'm not using systemd, I still live with well tested openrc;) 
This fork of EIO is a little more stable for me than others but it's not stable as it should. Under heavy load EIO do kaboom. Kernel throws panic and only way to catch trace is make photo or (probably) use RS232 serial cable. Are you willing to make changes to kernel module code or you are focused on improving helper scripts?

**Comment by elmystico** [comment]
Yes, no kernel code soon I guess, I won't find any spare time for that soon.
I think impossibleventures maybe will be willing to help.

**Comment by storcium** [comment]
Under heavy load for read or write?

Can I help test, or support the kernel effort?

Best,
Alex

**Comment by marcin-github** [comment]
I've got problem with stability of EIO under heavy writes.

**Comment by elmystico** [comment]
How's it going guys? Do you got solutions?
